scoala-online/main
# Feat/#16 implement mailing system

### This pull request solves issue:

Closes: #16 

### Which app is this issue related to?


### Types of changes


### Wiki Updates


### Testing checklist


#### Security Filters - **_For API Only_**

**_Please make sure to add the respective Security Filters before submitting a PR_**


#### Compatibility tests - **_For E-Learning React App Only_**


### What changes have been done within this issue?

I added dependencies for spring-boot-starter-mail and spring-boot-starter-thymeleaf in pom.xml.
I added a mail sender configuration together with templates for validation and reset password mails.
I added a service for sending mails.
I added the functionality for validating users and password resetting by changing the register endpoint and adding the following new endpoints:
- `verifyValidation` - validates a user based on the given code
- `requestResetPassword` - requests password reset
- `requestValidation` -  requests a new validation code
- `verifyResetPassword` - checks if the code for password reset is valid
- `resetPassword` - resets the password of an user based on the given code

### How should this be manually tested?

Firstly you need to create a new User entry using the /users/register path and then you will receive a mail with the validation code. Use the code in the /users/verify/validation/{code} path and the user will be validated. You can ask for a new validation code by using the /users/request/validation path.
You can also request a password reset by using the /users/request/reset_password path and you will receive a mail with the reset password code. Use the code in the /users/verify/reset_password/{code} path to check is the code is valid or use it in the /users/reset_password/{code} path and provide a new password that will replace the old one for the user with the given code. This code can expire.

### Screenshots or example usage
